pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -290,6 +305,11 @@ func (alloc *Action) allocateResourceForTasksWithTopology(tasks *util.PriorityQu
 		klog.V(4).InfoS("Find available hyperNodes for job", "jobName", job.UID, "tier", selectedTier, "hyperNodes", hyperNodes)
 	}
 	stmt, hyperNode := alloc.selectBestHyperNode(jobStmtsByTier[selectedTier], job)
+	if jobNewHyperNode, ok := LCAHyperNodeMap[hyperNode]; ok {
+		if jobNewHyperNode != hyperNodeJobAllocatedTo {
```

Comment:
How about just set annotation directly without if check as we will check whether update pg in closeSession finally.